pyaccel/intrabeam_scattering.py

```diff
@@ -0,0 +1,751 @@
+            return self._tau2
+        attr = 'tauy' if \
+            self.type_eqparams == self.EQPARAMS.RadIntegrals else 'tau2'
+        return getattr(self._eqpar_data, attr)
+
+    @tau2.setter
+    def tau2(self, val):
+        self._tau2 = float(val)
+        self._ibs_data = None
+
+    @property
+    def tau3(self):
+        """Return mode 3 (long) damping time to be used in calculations [s]."""
+        if self._tau3 is not None:
+            return self._tau3
+        attr = 'taue' if \
+            self.type_eqparams == self.EQPARAMS.RadIntegrals else 'tau3'
+        return getattr(self._eqpar_data, attr)
+
+    @tau3.setter
+    def tau3(self, val):
+        self._tau3 = float(val)
+        self._ibs_data = None
+
+    @property
+    def emit1(self):
+        """Emittance of mode 1 (x-plane) with IBS effects included [m.rad]."""
+        if self._ibs_data is not None:
+            return self._ibs_data['emit1'][-1]
+
+    @property
+    def emit2(self):
+        """Emittance of mode 2 (y-plane) with IBS effects included [m.rad]."""
+        if self._ibs_data is not None:
+            return self._ibs_data['emit2'][-1]
+
+    @property
+    def espread(self):
+        """Energy spread with IBS effects included."""
+        if self._ibs_data is not None:
+            return self._ibs_data['espread'][-1]
+
+    @property
+    def bunlen(self):
+        """Longitudinal bunch length with IBS effects included [m]."""
+        if self._ibs_data is not None:
+            return self._ibs_data['bunlen'][-1]
+
+    @property
+    def ibs_data(self):
+        """Return dictionary with IBS calculation data."""
+        if self._ibs_data is not None:
+            return self._ibs_data
+
+    def calc_ibs(self, print_progress=False):
+        """Calculate IBS effect on equilibrium parameters."""
+        gamma = self._acc.gamma_factor
+        beta_factor = self._acc.beta_factor
+        alpha = self._eqpar_data.etac
+        espread0 = self.espread0
+        bunlen0 = self.bunlen0
+        e10 = self.emit10
+        e20 = self.emit20
+        e30 = espread0 * bunlen0  # zero current longitudinal emittance
+
+        # We need to divide damping times by 2 because we are interested in
+        # emittance damping times, not on phase space variables damping times.
+        rate_sr_1 = 2 / self.tau1
+        rate_sr_2 = 2 / self.tau2
+        rate_sr_3 = 2 / self.tau3
+
+        num_part = self.particles_per_bunch
+        # Conversion factor from energy spread to bunch length, assuming a
+        # linear voltage (quadractic potential well). It will be used along
+        # the code to convert from longitudinal emittance to energy spread.
+        # This term is equal to w_s/eta_c/c
+        conv_sige2sigs = bunlen0 / espread0
+
+        # Copy values making them unique
+        spos, idx = _np.unique(self._optics_data.spos, return_index=True)
+        circum = spos[-1] - spos[0]
+        beta1 = self._optics_data.beta1[idx]
```

Comment:
This will raise an error if `optics_data` is a Twiss object. I handled this issue in `pyaccel.lifetime` with something like:

```
twi_names = [
            'betax', 'alphax', 'etax', 'etapx',
            'betay', 'alphay', 'etay', 'etapy']       
edteng_names = [
            'beta1', 'alpha1', 'eta1', 'etap1',
            'beta2', 'alpha2', 'eta2', 'etap2']
names = twi_names if self.type_optics == self.OPTICS.Twiss else edteng_names
beta1 = getattr(optics, names[0])[idx]
alpha1 = getattr(optics, names[1])[idx]
# and so on...
```

But you will have to add a property like `type_optics` to the user choose Twiss or EdwardsTeng type of optics calculation, just as is currently used to choose the beam equilibrium parameters in this class. For consistency in this case of IBS, I think I prefer to force the use of EdwardsTeng to avoid misleading results in studies with coupled lattices. Then you will not have to handle the Twiss or EdwardsTeng issues.

Reply:
I removed the option of using Twiss objects. This class will only deal with EdwardsTeng parametrization.